packages/shared/src/SliderLabels.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+
+import * as React from 'react';
+import { defaultTokens } from '@kiwicom/mobile-orbit';
+import { View } from 'react-native';
+import type { OnLayout } from '@kiwicom/mobile-shared';
+import type { TranslationType } from '@kiwicom/mobile-localization';
+
+import Text from './Text';
+import StyleSheet from './PlatformStyleSheet';
+
+type Props = {|
+  +start: {|
+    +label: TranslationType,
+    +value: number,
+  |},
+  +end?: {|
+    +label: TranslationType,
+    +value: number,
+  |},
+  +max: number,
+  +min: number,
+|};
+
+type State = {|
+  width: number,
+  labelStartWidth: number,
+  labelStartAtMax: boolean,
+  labelEndWidth: number,
+  paddingLeft: number,
+  paddingRight: number,
+|};
+
+export default class SliderLabels extends React.Component<Props, State> {
+  constructor() {
+    super();
+    this.state = {
+      width: 0,
+      labelStartWidth: 0,
+      labelStartAtMax: false,
+      labelEndWidth: 0,
+      paddingLeft: 0,
+      paddingRight: 0,
+    };
+  }
+
+  componentDidMount = () => {
+    requestAnimationFrame(this.setPadding);
+  };
+
+  getMaxPadding = () => {
+    const gap = 10;
+
+    if (this.props.end) {
+      return Math.floor(
+        this.state.width -
+          this.state.labelStartWidth -
+          this.state.labelEndWidth -
+          gap,
+      );
+    } else {
+      return Math.floor(this.state.width - this.state.labelStartWidth - gap);
+    }
+  };
+
+  calculateMarkerOffset = (type: 'start' | 'end'): number => {
```

Comment:
Why is this one function with a flag instead of

`calaculateMarkerStartOffset` and `calaculateMarkerEndOffset`?